### ISSUE TITLE
Implement support for MQTT protocol

### ISSUE THREAD

**Issue by rbroggi** [beatlabs/patron#456]
## Is your feature request related to a problem? Please describe

Include support for MQTT async protocol

**Comment by mantzas** [comment]
Could be interesting. The problem here is which library to use. MQTT v5 is not supported in a mature package last time we checked.
Can I also assume a client and not only a consumer?

**Comment by gkech** [comment]
We recently checked our options regarding Go MQTT clients and as @mantzas mentioned we found https://github.com/eclipse/paho.mqtt.golang is not MQTT v5 compatible. However, they do mention in their README that  https://github.com/eclipse/paho.golang is v5 compatible. It seems that it is a rather simple client, with a low star/fork count and also it is not as production-ready as https://github.com/hivemq/hivemq-mqtt-client which is used already by big smarthouse/automotive names.

**Comment by mantzas** [comment]
@gkech Yes I agree. We saw the Go MQTT v5 client and decided that it is not 100% mature and not a good choice. That's why our service that uses MQTT is written in Java (Spring Boot).

**Comment by gkech** [comment]
@mantzas Maybe it's not far fetched for us to develop a beatlabs mqtt client. 🚀

**Comment by mantzas** [comment]
@gkech this is a huge project and we most probably not have the capacity to do this. It is much easier to use either the v3 version or just use another language. My personal opinion.
